resources/lib/telstra_auth.py

```diff
@@ -122,8 +122,14 @@ def get_token(username, password):
                 continue
             data = offer.get('productOfferingAttributes')
             ph_no = [x['value'] for x in data if x['name'] == 'ServiceId'][0]
-    except Exception:
-        raise TelstraAuthException('Unable to determine eligible services')
+            if 'ph_no' not in locals():
```

Comment:
I don't like the use of locals() here. I usually like to define the var as None earlier on, then check it later with
```
  if ph_no:
    whatever
```

I'm curious though, would the line:
`ph_no = [x['value'] for x in data if x['name'] == 'ServiceId'][0]`
generate some error if ServiceId isn't found, about [0] not being valid for None or whatever?